snuba/datasets/__init__.py

```diff
@@ -74,6 +75,9 @@ def get_bulk_loader(self, source, dest_table):
         """
         raise NotImplementedError
 
+    def get_query_schema(self):
+        return QUERY_SCHEMA
```

Comment:
Should `__init__.py` return a `NotImplementedError` and the events dataset return `QUERY_SCHEMA`?

Or is `QUERY_SCHEMA` for more than just `events`?

Reply:
This is an interesting point, and I think you're onto something here — at least right now, the `events` dataset is the only dataset that can use the query schema as currently defined (reliably at least), so it probably makes sense to only expose it as part of that class. That would at least make it obvious that other datasets don't support querying through the API yet, which I think is a worthwhile change.

Longer-term, I see there eventually being a generic query schema that *is* supported by all datasets and would be exposed here on the dataset (or wherever this method ends up over the next few days of changes to support that.) That basic schema will be able to be augmented/extended to add dataset specific functionality, but I think all datasets will share a common subset of valid parameters — probably things that are just generically analogous to query clauses (select columns, conditions, group by, order by, etc.)

Reply:
Made this change in 2f027578b7f78f2898c5e3fec2d0b06266543378.